README.md

```diff
@@ -53,6 +158,20 @@ combined into one to avoid having to keep an actual queue and send multiple
 messages. The same process occurs when the client receives a block and sends a
 cancel message for it.
 
+### Sessions
+
+Sessions track related requests for blocks, and attempt to optimize transfer speed and reduce the number of duplicate blocks sent across the network. The basic optimization of sessions is to limit asks for blocks to the peers most likely to have that block and most likely to respond quickly. This is accomplished by tracking who responds to each block request, and how quickly they respond, and then taking optimizing future requests with that data. Ideally, you want to ask peers who've already responded, and responded quickly, and distribute requests among them to prevent duplicates.
```

Comment:
```suggestion
Sessions track related requests for blocks, and attempt to optimize transfer speed and reduce the number of duplicate blocks sent across the network. The basic optimization of sessions is to limit asks for blocks to the peers most likely to have that block and most likely to respond quickly. This is accomplished by tracking who responds to each block request, and how quickly they respond, and using that information to optimize future requests. Ideally, you want to ask peers who've already responded, and responded quickly, and distribute requests among them to avoid receiving many duplicate blocks.
```